linkerd/app/core/src/telemetry/build_info.rs

```diff
@@ -0,0 +1,63 @@
+use linkerd2_metrics::{metrics, FmtLabels, FmtMetric, FmtMetrics, Gauge};
+use std::fmt;
+use std::string::String;
+use std::sync::Arc;
+
+const GIT_BRANCH: &'static str = env!("GIT_BRANCH");
+const GIT_VERSION: &'static str = env!("GIT_VERSION");
+const RUST_VERSION: &'static str = env!("RUST_VERSION");
+
+metrics! {
+    proxy_build_info: Gauge {
+        "Proxy build info"
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct Report {
+    name: String,
+    // `value` remains constant over the lifetime of the proxy so that
+    // build information in `labels` remains accurate
+    value: Arc<Gauge>,
+    labels: Arc<BuildInfoLabels>,
+}
+
+#[derive(Clone, Debug, Default)]
+struct BuildInfoLabels {
+    git_branch: String,
+    git_version: String,
+    rust_version: String,
```

Comment:
could be worth adding a label for whether the proxy was built in debug mode or release mode (which I believe is [available in an env var called `PROFILE`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts)).